include/drivers/adc.h

```diff
@@ -144,6 +144,50 @@ struct adc_channel_cfg {
 #endif /* CONFIG_ADC_CONFIGURABLE_INPUTS */
 };
 
+#define ADC_CHANNEL_CFG_DT(node_id) (struct adc_channel_cfg) { \
+	.channel_id       = DT_REG_ADDR(node_id), \
+	.gain             = DT_STRING_TOKEN(node_id, gain), \
+	.reference        = DT_STRING_TOKEN(node_id, reference), \
+	.acquisition_time = DT_PROP(node_id, acquisition_time), \
+	.differential     = DT_PROP(node_id, differential), \
+IF_ENABLED(CONFIG_ADC_CONFIGURABLE_INPUTS, \
+	(.input_positive  = DT_PROP(node_id, input_positive), \
+	 .input_negative  = DT_PROP_OR(node_id, input_negative, 0),)) \
+}
+
+#define ADC_CHANNEL_DT_NODE(adc_node, chan_id) \
+	DT_CHILD(adc_node, DT_CAT(channel_, chan_id))
```

Comment:
The DT_CAT* macros are for internal use within the devicetree.h header and sub-headers; please use something from util_macro.h or roll your own concatenation helpers.

Reply:
Used `UTIL_CAT` instead.